pkg/kev/config/extensions_test.go

```diff
@@ -124,4 +165,43 @@ var _ = Describe("Extentions", func() {
 		})
 	})
 
+	Describe("Merge", func() {
+		It("merges target into base", func() {
+			k8sBase := config.DefaultK8SConfig()
+			var k8sTarget config.K8SConfiguration
+			k8sTarget.Workload.Replicas = 10
+			k8sTarget.Workload.LivenessProbe.Type = config.ProbeTypeNone.String()
+
+			expected := k8sBase
+			expected.Workload.Replicas = 10
+			expected.Workload.LivenessProbe.Type = config.ProbeTypeNone.String()
+
+			result, err := k8sBase.Merge(k8sTarget)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(BeEquivalentTo(expected))
+		})
+
+		When("results in default config are empty/invalid", func() {
```

Comment:
The bodies of two cases in this context are practically the same.. I think it'd make sense to turn the below into two separate context and set the initial condition for the test cases within those contexts. 

For the first one, how about the following format?

```go
When("k8s extension map is empty", func() {
    BeforeEach(func() {
        extensions := make(map[string]interface{})
        parsedConf, err := config.K8SCfgFromMap(extensions)
        Expect(err).NotTo(HaveOccurred())
    })

    It("doesn't return an error and resultant configuration matches the default configuration", func() {
        result, err := config.DefaultK8SConfig().Merge(parsedConf)
        Expect(err).NotTo(HaveOccurred())
        Expect(result).To(BeEquivalentTo(config.DefaultK8SConfig()))
    })
})
```

and the case after that could follow the same format for consistency.